_posts/en/newsletters/2019-11-13-newsletter.md

```diff
@@ -0,0 +1,360 @@
+    response was that fewer restrictions will allow a later soft fork to
+    define other uses for v1 scriptPubKeys with shorter or longer
+    lengths.  In the meantime, if taproot is adopted, those shorter and
+    longer v1 scriptPubKeys will be spendable by anyone (as they are
+    today).
+
+    This started a discussion among experts about a recently-discovered
+    issue related to how this flexibility interacts with a somewhat
+    older [issue][bech32 length change] discovered in the bech32 address
+    encoding algorithm.  [BIP173][] bech32 addresses are supposed to
+    guarantee that up to four errors will be detected in an incorrectly
+    copied address and that only about one miscopied address in a
+    billion that had five or more errors would go undetected.
+    Unfortunately, those calculations were made under the assumption
+    that the length of the copied address would be the same as the
+    original.  If the copied address is instead longer or shorter,
+    bech32 can fail to detect even a single-character error on rare
+    occasions.
+
+    For existing P2WPKH and P2WSH bech32 addresses, this is very
+    unlikely to be an issue since the restriction that v0
+    scriptPubKeys be exactly 22 or 34 bytes means that a miscopied
+    P2WPKH address would need to either contain an extra 12 bytes or a
+    P2WSH address would need to omit 12 bytes, meaning a user would need
+    to type about 19 extra or fewer bech32 characters---an awfully big
+    mistake.  <!-- 8 bits per byte * 12 bytes / 5 bits per bech32
+    character = 19.2 bech32 characters -->
+
+    But if P2TR is only defined for 34-byte v1 scriptPubKeys and it
+    remains the case that anyone can spend 33-byte and 35-byte v1
+    scriptPubKeys, then it would be possible for a user to make a
+    single-character mistake and lose all of the money they intended to
+    spend.  Author of both BIP173 and the taproot proposal Pieter Wuille
+    [posted][wuille bech32 workaround] to the Bitcoin-Dev mailing some
+    options for addressing the problem and requested feedback on what
+    options people would prefer to see implemented.  One option would be
+    restricting all current bech32 implementations to rejecting any
+    native segwit addresses that don't result in a 22 or 34 byte
+    scriptPubKey.  Then an upgraded version of bech32 could later be
+    developed with better detection of inserted or deleted characters.
+
+    Many other less critical discussions also resulted from the week's
+    review of taproot, and discussion logs are available ([1][tr meet1],
+    [2][tr meet2]) for anyone interested in the discussion details.
+
+    {:#x-only-pubkeys}
+    In other schnorr/taproot news, Jonas Nick published an [informative
+    blog post][x-only pubkeys] about a recent major change to
+    [bip-schnorr][] and [bip-taproot][] that reduced the size of
+    serialized public keys from 33 bytes to 32 bytes without reducing
+    security.
```

Comment:
James Chiang's idea suggested to Pieter Wuille on IRC :1st_place_medal:

Reply:
Thanks @jonatack! It was originally a conversation with John, who later expanded it to include Pieter. John did give me credit for reducing to 32B public keys here: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2019-May/016943.html